tools/db_import/db_import/rules.py

```diff
@@ -0,0 +1,100 @@
+## Copyright 2023 The OpenXLA Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+import csv, datetime, io, json, re, os
+from typing import Any, Optional, Callable
+import rjsonnet
+from google.cloud import storage
+from db_import.utils import first_no_except
+
+
+class BenchmarkRunAlreadyPresentError(Exception):
+  pass
+
+
+# Applies the given `rule`` to `file`.
+# - Returns False if `rule[filepath_regex]` does not match the filepath of `file`.
+# - If it matches, it evaluates the JSONNet expression `rule[result]` and returns its result (parsed as JSON).
+def apply_rule_to_file(
+    rule: dict,
+    file: storage.Blob,
+    config: dict,
+    snippets: dict[str, str],
+    presence_check: Optional[Callable[[Any, Any], bool]],
+    dump_files_to: Optional[str],
```

Comment:
Can we use `pathlib.Path` for this parameter? So we can have its convenient methods below (https://docs.python.org/3/library/pathlib.html#correspondence-to-tools-in-the-os-module)